src/utils/trust/trust.involved.ts

```diff
@@ -0,0 +1,208 @@
+import { mapTrustWhoIsInvolvedToPage } from '../../utils/trust/who.is.involved.mapper';
+import { FormattedValidationErrors, formatValidationError } from '../../middleware/validation.middleware';
+import { IndividualTrustee, TrustHistoricalBeneficialOwner } from '../../model/trust.model';
+import { getIndividualTrusteesFromTrust, getFormerTrusteesFromTrust } from '../../utils/trusts';
+import {
+  CHECK_YOUR_ANSWERS_URL,
+  REGISTER_AN_OVERSEAS_ENTITY_URL,
+  TRUST_INVOLVED_PAGE,
+  UPDATE_AN_OVERSEAS_ENTITY_URL,
+  UPDATE_CHECK_YOUR_ANSWERS_URL,
+  UPDATE_TRUSTS_INDIVIDUALS_OR_ENTITIES_INVOLVED_PAGE,
+  UPDATE_TRUSTS_TELL_US_ABOUT_IT_URL,
+  UPDATE_TRUSTS_ASSOCIATED_WITH_THE_OVERSEAS_ENTITY_URL
+} from '../../config';
+
+const TRUST_INVOLVED_TEXTS = {
+  title: 'Individuals or entities involved in the trust',
+  boTypeTitle: {
+    [BeneficialOwnerTypeChoice.individual]: 'Individual beneficial owner',
+    [BeneficialOwnerTypeChoice.otherLegal]: 'Other legal beneficial owner',
+  },
+  trusteeTypeTitle: {
+    [TrusteeType.HISTORICAL]: 'Former beneficial owner',
+    [TrusteeType.INDIVIDUAL]: 'Individual',
+    [TrusteeType.LEGAL_ENTITY]: 'Legal entity',
+  },
+};
+
+type TrustInvolvedPageProperties = {
+  backLinkUrl: string,
+  templateName: string;
+  pageParams: {
+    title: string;
+  },
+  pageData: {
+    beneficialOwnerTypeTitle: Record<string, string>;
+    trusteeTypeTitle: Record<string, string>;
+    trusteeType: typeof TrusteeType;
+    individualTrusteeData: IndividualTrustee[];
+    formerTrusteeData: TrustHistoricalBeneficialOwner[];
+    checkYourAnswersUrl: string;
+    beneficialOwnerUrlDetach: string;
+    trustData: CommonTrustData,
+  } & TrustWhoIsInvolved,
+  formData?: TrustWhoIsInvolvedForm,
+  errors?: FormattedValidationErrors,
+  url: string,
+};
+
+const getPageProperties = (
+  req: Request,
+  isUpdate: boolean,
+  formData?: TrustWhoIsInvolvedForm,
+  errors?: FormattedValidationErrors,
+): TrustInvolvedPageProperties => {
+  const trustId = req.params[config.ROUTE_PARAM_TRUST_ID];
+
+  const appData = getApplicationData(req.session);
+
+  return {
+    backLinkUrl: getBackLink(isUpdate, trustId),
+    templateName: getTemplateName(isUpdate),
+    pageParams: {
+      title: TRUST_INVOLVED_TEXTS.title,
+    },
+    pageData: {
+      trustData: mapCommonTrustDataToPage(appData, trustId),
+      ...mapTrustWhoIsInvolvedToPage(appData, trustId),
+      beneficialOwnerTypeTitle: TRUST_INVOLVED_TEXTS.boTypeTitle,
+      trusteeTypeTitle: TRUST_INVOLVED_TEXTS.trusteeTypeTitle,
+      individualTrusteeData: getIndividualTrusteesFromTrust(appData, trustId),
+      formerTrusteeData: getFormerTrusteesFromTrust(appData, trustId),
+      trusteeType: TrusteeType,
+      checkYourAnswersUrl: getCheckYourAnswersUrl(isUpdate),
+      beneficialOwnerUrlDetach: `${config.TRUST_ENTRY_URL}/${trustId}${config.TRUST_BENEFICIAL_OWNER_DETACH_URL}`,
```

Comment:
I can't find any use of beneficialOwnerUrlDetach anywhere, maybe not looking in the right places but do we need an Update equivalent of this URL as well? As that is Registration specific

Reply:
Yes I was wondering about this. Like yourself I couldn't find what it's used for, but I'll make an update equivalent to keep it consistent

Reply:
Fixed

Reply:
Yeah but what you have added there now is a URL to a route that doesn't exist for Update, so that's quite right either. (As we don't know what that URL in registration does and haven't added anything similar in Update). Have a look around registration routes and try to see what that existing URL in registration is used for

Reply:
The page of the url for registration doesn't seem to be completed yet, will revert back to what it was originally
<img width="865" alt="Screenshot 2023-07-18 at 15 20 52" src="https://github.com/companieshouse/overseas-entities-web/assets/120725571/65e40a88-c0d8-47a3-aec1-491acc3a4d38">

Reply:
Reverted now

Reply:
yeah I think thats good, whenever that is implemented (if it ever is) then something can be done for update too. No point adding in another route that does nothing just for the sake of it. Nice!